tests/backends/tango/test_dsr.py

```diff
@@ -90,36 +43,64 @@ def test_status(self, tango_context):
         expected = "The device is in ON state."
         assert tango_context.command_inout("Status") == expected
 
-    def test_read_int(self, client_read):
-        client_read(["ReadInt"], AttrR(Int())._value)
-
-    def test_read_write_int(self, client_read, client_write):
-        client_read(["ReadWriteInt"], AttrR(Int())._value)
-        client_write(["ReadWriteInt"], AttrR(Int())._value)
-
-    def test_read_write_float(self, client_read, client_write):
-        client_read(["ReadWriteFloat"], AttrR(Float())._value)
-        client_write(["ReadWriteFloat"], AttrR(Float())._value)
-
-    def test_read_bool(self, client_read):
-        client_read(["ReadBool"], AttrR(Bool())._value)
-
-    def test_write_bool(self, client_write):
-        client_write(["WriteBool"], AttrR(Bool())._value)
-
-    def test_string_enum(self, client_read, client_write):
-        enum = AttrR(String(), allowed_values=["red", "green", "blue"])._value
-        client_read(["StringEnum"], enum)
-        client_write(["StringEnum"], enum)
-
-    def test_big_enum(self, client_read):
-        client_read(["BigEnum"], AttrR(Int(), allowed_values=list(range(1, 18)))._value)
-
-    def test_go(self, client_exec):
-        client_exec(["Go"])
-
-    def test_read_child1(self, client_read):
-        client_read(["SubController01", "ReadInt"], AttrR(Int())._value)
-
-    def test_read_child2(self, client_read):
-        client_read(["SubController02", "ReadInt"], AttrR(Int())._value)
+    def test_read_int(self, assertable_controller, tango_context):
+        expect = 0
+        with assertable_controller.assert_read_here(["read_int"]):
+            result = tango_context.read_attribute("ReadInt").value
+        assert result == expect
+
+    def test_read_write_int(self, assertable_controller, tango_context):
+        expect = 0
+        with assertable_controller.assert_read_here(["read_write_int"]):
+            result = tango_context.read_attribute("ReadWriteInt").value
+        assert result == expect
+        with assertable_controller.assert_write_here(["read_write_int"]):
+            tango_context.write_attribute("ReadWriteInt", expect)
```

Comment:
It might be good if this did a read, a write to as different value and then another read to check it has changed

Reply:
OK will add